### DISCUSSION TITLE
Add additional Hard Drives to increase the /Storage mount

### DISCUSSION THREAD

**Discussion by davedoc1984** [retronas/discussions#86] [Q&A]
Probably a Daft Question, Can I add additional Hard Drives to increase the /Storage mount so that MiSTer etc can use both hard drives?

**Comment by sairuk** [comment]
I believe right now, you would need to do something like LVM where the physical volumes (hard disks) are joined into a pool of disks and then the partition is striped over multiple drives.

We have a want for this in our [Ideas](https://github.com/danmons/retronas/wiki/Ideas) page, but no ETA on support at the moment.

There is no reason you could not do this manually for the time being though and mount the LV in place where your top level directory currently is.

**Reply by lilylarceny** [reply]
@sairuk my 18tb drive is filling up pretty quickly, being able to at least set a root directory for different systems, for instance xbox 360 on one path and ps3 on another path would really help a lot. I have my 18tb formatted to NTFS. there seems to be some debate on the internet about if it's a great idea to convert NTFS to LVM or not, but ultimately I would like to be able to connect my drive to my windows desktop for easy file transfer.

**Reply by birdybro** [reply]
LVM is essentially already supported in Windows as well, funnily enough. It's called "Storage Spaces" --> https://support.microsoft.com/en-us/windows/storage-spaces-in-windows-b6c8b540-b8d8-fb8a-e7ab-4a75ba11f9f2

**Reply by lilylarceny** [reply]
However, I am on a Debian box, and would like to be able easily disconnect any of the drives and connect them to a windows PC for file transfer. That seems super difficult if it's part of an LVM on a Debian system? Being able to specify a path for each game console's share would probably be a lot less complicated. Maybe I am wrong here, but I don't think including the drive in an LVM would make it easy to disconnect it and reconnect it to other computers.

**Reply by sairuk** [reply]
I have added your request about system based data paths to the [Ideas](https://github.com/danmons/retronas/wiki/Ideas) page so we can look at what its going to take to do this architecturally. I have some ideas, but they'll be down the list for a while.

Once you move outside the current offering things are going to get very manual/complicated for the moment, that can't be helped. We'll just be throwing around ideas here for solutions against the current offering

Note this next idea(s) is/are completely theoretical,

If you don't want to go with/wait for LVM another option may be to custom mount each of your "system drives" in place into the RetroNAS structure through fstab, automount or systemd automount units. e.g. you move your global data path back to smaller storage, e.g SD card or USB then you mount say a Microsoft drive to /data/retronas/roms/microsoft and let RetroNAS create its structure. You could arguably achieve multiple systems on a single drive through partitioning, with parition 1 being microsoft, partition 2 being sony etc etc

I'd expect you'd need to run the romdirs creation twice, once to create the top level microsoft directory, then mount your drive in place to that top level dir and then again to create the RetroNAS directory structure on the mounted drive.

A second idea was to mount you drives wherever and symlink across into main structure but I think that will currently either cause errors during playbook runs or the link could get removed and replaced with a directory as per our requirements. I'd need to test that thought since I don't think we are forcing anything there so it should just return an error and not take any action.

Happy for others to chime in on the general storage side of things, again tho I have added this to the [Ideas](https://github.com/danmons/retronas/wiki/Ideas) page for future consideration.
